robot_smach_states/src/robot_smach_states/manipulation/grab.py

```diff
@@ -176,15 +178,14 @@ def execute(self, userdata=None):
                 return 'failed'
 
         # Pre-grasp --> this is only necessary when using visual servoing
-        # rospy.loginfo('Starting Pre-grasp')
-        # if not arm.send_goal(goal_bl.x, goal_bl.y, goal_bl.z, 0, 0, 0,
-        #                      frame_id=self.robot.base_link_frame',
-        #                      timeout=20, pre_grasp=True, first_joint_pos_only=True
-        #                      ):
-        #     rospy.logerr('Pre-grasp failed:')
-        #     arm.reset()
-        #     arm.gripper.send_goal('close', timeout=None)
-        #     return 'failed'
+        if self._use_visual_servoing:
+            rospy.loginfo('Starting Pre-grasp')
+            if not arm.send_goal(goal_bl.x, goal_bl.y, goal_bl.z, 0, 0, 0, frame_id=self.robot.base_link_frame,
+                                 timeout=20, pre_grasp=True, first_joint_pos_only=True):
+                rospy.logerr('Pre-grasp failed:')
+                arm.reset()
+                arm.gripper.send_goal('close', timeout=None)
+                return 'failed'
```

Comment:
This appears to do exactly what is being done below here. Which is move the arm to the goal pose. Visual servoing does not appear to be happening here.
Question 1: How should the behaviour of the robot change with this flag.
Question 2: was this change tested.

Reply:
This was first commented out, but indeed if it does exactly what we do below we can remove it since it was not used before aswell.